bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/ComponentUtils.java

```diff
@@ -62,6 +65,30 @@ public class ComponentUtils {
     private static final String EDGE_DELIVERY_FRAGMENT_CONTAINER_REL_PATH = "root/section/form";
     private static final String[] EDGE_DELIVERY_RESOURCE_TYPES = new String[] { "core/franklin/components/page/v1/page" };
     private static final Logger logger = LoggerFactory.getLogger(ComponentUtils.class);
+    private static final Map<String, List<String>> SUBMIT_ACTIONS_CACHE = new ConcurrentHashMap<>();
+    private static final Map<String, Long> CACHE_TIMESTAMPS = new ConcurrentHashMap<>();
+    private static final long CACHE_TTL = TimeUnit.HOURS.toMillis(24);
+    private static final String CACHE_KEY = "submit_actions";
+
+    private static List<String> getFromCache() {
```

Comment:
Please create separate class for cache, let's try to not pollute a single file for all use-cases